### DISCUSSION TITLE
Remove guest user access from mobile app

### DISCUSSION THREAD

**Discussion by nidhikachhawa24** [Sunbird-Ed/Community#235] [General Discussion]
I am creating an app for a tenant and I want to allow users to access tenant data only when the user is logged in.

I don't want users to have guest user access in mobile app.
**Reason:** They can explore all the content of all the tenants. This should should not be allowed.

**How we can remove that from mobile app?**

cc: @swayangjit @vpPavithra @luckynvdu @baljeetsingh

**Comment by nidhikachhawa24** [comment]
@swayangjit can you please help in achieving the above scenario?

**Reply by Krishnaj20** [reply]
@nidhikachhawa24 - I believe if the first screen in the app asks the user to login/register to continue which should address this issue?

@swayangjit

**Reply by swayangjit** [reply]
@Krishnaj20 Yes . For admin usecase we are asking for login once you click on that user type.

**Reply by nidhikachhawa24** [reply]
@swayangjit @Krishnaj20 

Sorry, I didn't understand. Can you explain a bit more?

Also I'm sharing the flow of the mobile app 
**Here is the scenario:**
1. When the user opens the app, this should be the flow -> splash screen -> role selection -> login screen
2. When the user has not logged in and kill the app -> open app again -> splash screen -> login screen
3. When user logout -> redirect it to login screen

**Only registered users will be able to able to access the app so we don't need the guest user.**

**Reply by swayangjit** [reply]
@nidhikachhawa24  Same thing has been implemented for userType admin/Head Teacher/ Officials.
Once you select userType it will navigate it to the login screen but once you logout it will navigate again back to userType selection page.
